src/core/grid_based_algorithms/lbgpu_cuda.cu

```diff
@@ -4573,6 +4590,7 @@ void lb_integrate_GPU() {
                lb_boundary_velocity, lb_boundary_force);
   }
 #endif
+  philox_counter += 1;
```

Comment:
Just use the current timestep number here instead of adding a global static variable...

Reply:
that increases coupling

Reply:
Yours increases the number of globals. Both are bad.

The advantage of using the time step is that you can use `integrate(0)` and it will produce the same LB forces every time. Also, you can then calculate the missing LB forces in the first time step (the ones that we currently print a warning about). Same after loading a checkpoint, if you checkpoint the time step number.

Reply:
Since there is a plan to refactor the LB we want to have it as isolated as possible. During refactoring the globals will go, the coupling to the MD core won't.

Reply:
Also it's not hard to checkpoint a single number...

Reply:
How do you want to deal with `integrate(0)`? That shouldn‘t advance the counter.

Reply:
Sure, there is an interface missing that just calculatesforces from the fluid without propagating it. Whats your point?

Reply:
You wouldn‘t need a separate function if you use the time step directly. You have introduced a variable here that will always need to be equal to the time step number. That kind of design is just as bad as the increased coupling: If you don‘t directly use the time step number, you always risk the two numbers diverging, e.g. if you forget to take care of a special code path that doesn‘t increase the time step number.

Reply:
The place that calls `lb_integrate_GPU()` has access to the `sim_time` variable. Just pass it in from there, then you don‘t need to increase coupling.